src/modules/mavlink/streams/ADSB_VEHICLE.hpp

```diff
@@ -103,6 +128,89 @@ class MavlinkStreamADSBVehicle : public MavlinkStream
+
+			// Required update for dynamic message is 5 [Hz]
+			mavlink_uavionix_adsb_out_dynamic_t dynamic_msg = {
+				.utcTime = UINT32_MAX,
+				.gpsLat = INT32_MAX,
+				.gpsLon = INT32_MAX,
+				.gpsAlt = INT32_MAX,
+				.baroAltMSL = INT32_MAX,
+				.accuracyHor = UINT32_MAX,
+				.accuracyVert = UINT16_MAX,
+				.accuracyVel = UINT16_MAX,
+				.velVert = INT16_MAX,
+				.velNS = INT16_MAX,
+				.VelEW = INT16_MAX,
+				.state = UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND,
+				.squawk = static_cast<uint16_t>(_adsb_squawk.get()),
+				.gpsFix = 0,
+				.numSats = UINT8_MAX,
+				.emergencyStatus = static_cast<uint8_t>(_adsb_emergc.get())
+			};
+
+			vehicle_status_s vehicle_status;
+
+			if (_vehicle_status_sub.update(&vehicle_status)) {
+				if (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
+					dynamic_msg.state |= ~UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND;
+				}
+			}
+
+			sensor_gps_s vehicle_gps_position;
+
+			if (_vehicle_gps_position_sub.update(&vehicle_gps_position)) {
+
+				dynamic_msg.utcTime = static_cast<uint32_t>(vehicle_gps_position.time_utc_usec / 1000000ULL);
+				dynamic_msg.gpsLat = vehicle_gps_position.lat;
+				dynamic_msg.gpsLon = vehicle_gps_position.lon;
+				dynamic_msg.gpsAlt = vehicle_gps_position.alt;
+				dynamic_msg.gpsFix = vehicle_gps_position.fix_type;
+				dynamic_msg.numSats = vehicle_gps_position.satellites_used;
+				dynamic_msg.accuracyHor = static_cast<uint32_t>(vehicle_gps_position.eph * 1000.0f); // convert [m] to [mm]
+				dynamic_msg.accuracyVert = static_cast<uint16_t>(vehicle_gps_position.epv * 100.0f); // convert [m] to [cm]
+				//_dynamic_data.accuracyVel //TODO: anything to set here?
+				dynamic_msg.velVert = static_cast<int16_t>(vehicle_gps_position.vel_d_m_s * 100.0f); // convert [m/s] to [cm/s]
+				dynamic_msg.velNS = static_cast<int16_t>(vehicle_gps_position.vel_n_m_s * 100.0f); // convert [m/s] to [cm/s]
+				dynamic_msg.VelEW = static_cast<int16_t>(vehicle_gps_position.vel_e_m_s * 100.0f); // convert [m/s] to [cm/s]
+			}
+
+			vehicle_air_data_s vehicle_air_data;
+
+			if (_vehicle_air_data_sub.update(&vehicle_air_data)) {
+				dynamic_msg.baroAltMSL = static_cast<int32_t>(vehicle_air_data.baro_pressure_pa / 100.0f); // convert [Pa] to [mBar]
+			}
+
+			mavlink_msg_uavionix_adsb_out_dynamic_send_struct(_mavlink->get_channel(), &dynamic_msg);
+
+			_period_counter++;
+
+			if (_period_counter >= CFG_PERIOD_COUNTER) {
+
+				// Required update for static message is 0.1 [Hz]
+				mavlink_uavionix_adsb_out_cfg_t cfg_msg = {
+					.ICAO = static_cast<uint32_t>(_adsb_icao.get()),
+					.stallSpeed = 0, //TODO: any existing param?
+					//memcpy(cfg_msg.callsign, "PX4 UAV ", sizeof(cfg_msg.callsign)); TODO: shall to be changeable
+					.emitterType = static_cast<uint8_t>(_adsb_emit_type.get()),
+					.aircraftSize = static_cast<uint8_t>(_adsb_len_width.get()),
+					.gpsOffsetLat = 0x0, //TODO: add missing param
```

Comment:
could be https://docs.px4.io/main/en/advanced_config/parameter_reference.html#EKF2_GPS_POS_Y

Reply:
Nice suggestion!

Reply:
actually needde to add a parameter since it is an enumeration.